Assets/MixedRealityToolkit.Examples/Demos/SpatialAwareness/Scripts/DemoSpatialMeshHandler.cs

```diff
@@ -34,13 +34,12 @@ private IMixedRealitySpatialAwarenessSystem SpatialAwarenessSystem
         /// </summary>
         private Dictionary<int, uint> meshUpdateData = new Dictionary<int, uint>();
 
-        private async void OnEnable()
+        private void Start()
         {
-            await new WaitUntil(() => SpatialAwarenessSystem != null);
-            SpatialAwarenessSystem.RegisterHandler<SpatialAwarenessHandler>(this);
+            SpatialAwarenessSystem?.RegisterHandler<SpatialAwarenessHandler>(this);
         }
 
-        private void OnDisable()
+        private void OnDestroy()
```

Comment:
This may be more an open ended question, but does it make sense to stay registered and continue listening to / handling events even when disabled?

Reply:
good question...